File: lib/src/domain/models/query_grouping.dart
import 'package:equatable/equatable.dart';

import 'package:query_wizard/domain.dart';

class QueryGrouping extends Equatable {
  const QueryGrouping({
    required this.name,
    required this.type,
    this.parent,
    required this.elements,
  });

  final String name;
  final QueryGroupingType type;
  final QueryGrouping? parent;
  final List<QueryGrouping> elements;

  @override
  List<Object> get props => [
        name,
        type,
        parent ?? '',
        elements,
      ];

  @override
  String toString() {
    return '$name $type';
  }
}

File: server/src/QueryWizard.Domain/src/Entities/QueryGrouping.cs
using System.Collections.Generic;
using QueryWizard.Domain.Enums;

namespace QueryWizard.Domain.Entities
{
    public class QueryGrouping
    {
        public string Id { get; set; }

        public QueryElement Field { get; set; }

        public QueryGroupingType Type { get; set; }

        public QueryGrouping Parent { get; set; }

        public List<QueryGrouping> Elements { get; set; }
    }
}
